src/Google/GlobalSiteTag.php

```diff
@@ -66,18 +83,42 @@ public function register(): void {
 
 		add_action(
 			'wp_head',
-			function() use ( $ads_conversion_id ) {
+			function () use ( $ads_conversion_id ) {
 				$this->activate_global_site_tag( $ads_conversion_id );
 			},
 			999998
 		);
 		add_action(
-			'wp_head',
-			function() use ( $ads_conversion_id, $ads_conversion_label ) {
-				$this->maybe_display_event_snippet( $ads_conversion_id, $ads_conversion_label );
+			'woocommerce_before_thankyou',
+			function ( $order_id ) use ( $ads_conversion_id, $ads_conversion_label ) {
+				$this->maybe_display_conversion_and_purchase_event_snippets( $ads_conversion_id, $ads_conversion_label, $order_id );
 			},
 			1000000
 		);
+
+		add_action(
+			'woocommerce_after_single_product',
+			function () {
+				$this->display_view_item_event_snippet();
+			}
+		);
+
+		add_action(
+			'wp_body_open',
+			function () {
+				$this->display_page_view_event_snippet();
+				$this->display_cart_page_snippet();
+			}
+		);
+
+		add_filter(
+			'wc_add_to_cart_message_html',
```

Comment:
This hook works when we add a product to the cart from a single product page. Both when we have redirect to cart enable and when it's disabled. But when we have redirect to cart disabled and Ajax add to cart enabled, we don't get this notification, so the add to cart event is never tracked.

![image](https://user-images.githubusercontent.com/11388669/166675748-aaa371c9-b1c9-4455-8d9e-b02037376a44.png)

This is the default configuration for WooCommerce so there are quite a few stores who will have it set like this, so I think we should consider adding the tracking based on when the button is clicked, like is done in the [Google Analytics Integration](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/1.5.8/includes/class-wc-google-analytics-js.php#L509-L515) (although that would need to be converted from jQuery to regular JavaScript).

Reply:
Thanks so much for getting back to me. I saw you're working on this [RP](https://github.com/woocommerce/google-listings-and-ads/pull/1488) Will this PR help track the "add to cart" action on single product page and multiple product page with "add to cart" buttons?

Reply:
Right now the intention of that PR is to only work with the Gutenberg blocks. So it only works with a block version of an archive page which is created using the All Products block. It doesn't do anything for traditional archive pages, or single product pages.
However it might be a good idea to include that there as well. Since it loads a JS file which can react to specific events, it might be better to look for add_to_cart buttons on those pages and then track the event. Which is probably a better solution then injecting a script for each add_to_cart button. Let me know if you think that's a better approach and I can work on adding that to the PR.

What I do need clarified though, is why there is a difference between using `name` or `item_name` fields in the docs and PRD. It doesn't look like the item details get tracked correctly so I'll need some clarification on what fields need to get sent in the event.

Reply:
Thanks for your reply. It will be better to add tracking to each "add to cart" button. And I've updated the name in the list of items to item_name.

Reply:
Moving the add_to_cart handling to your PR[https://github.com/woocommerce/google-listings-and-ads/pull/1488](url) sounds good to me. I'll remove this "add to cart" tracking part.